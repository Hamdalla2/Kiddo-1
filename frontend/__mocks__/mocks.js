import 'react-native-gesture-handler/jestSetup';
require('jest-fetch-mock').enableMocks();

if(window){
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
}
jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};

  return Reanimated;
});

// Silence the warning: Animated: `useNativeDriver` is not supported because the native animated module is missing
jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');