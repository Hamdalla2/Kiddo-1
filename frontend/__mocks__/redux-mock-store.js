  
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore(middlewares);

export default mockStore;