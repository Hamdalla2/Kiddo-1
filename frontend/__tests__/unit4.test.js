/**
* @jest-environment jsdom
*/

import '../__mocks__/mocks';
import React from 'react'
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { act } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import '@react-native-community/async-storage/jest/async-storage-mock';
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
import Art from '../screens/art';

test('renders correctly', async () => {
  const test = render(<Provider store={mockStore({})}><Art/></Provider>).toJSON();
  await act(async () => {expect(test).toMatchSnapshot();})
});