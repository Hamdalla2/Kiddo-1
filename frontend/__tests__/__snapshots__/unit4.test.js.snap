// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-view-1dbjc4n"
>
  <div
    className="css-view-1dbjc4n r-paddingBottom-97e31f r-paddingLeft-bv2aro r-paddingRight-hxflta r-paddingTop-10xqauy"
  >
    <div
      className="css-view-1dbjc4n r-WebkitOverflowScrolling-150rngu r-flexDirection-eqz5dr r-flexGrow-16y2uox r-flexShrink-1wbh5a2 r-overflowX-11yh6sk r-overflowY-1rnoaur r-transform-1sncvnh"
      onScroll={[Function]}
      onTouchMove={[Function]}
      onWheel={[Function]}
    >
      <div
        className="css-view-1dbjc4n"
      >
        <div
          className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
          data-focusable={true}
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-view-1dbjc4n r-backgroundColor-8bo4q2 r-borderRadius-kdyh1x r-boxShadow-l8kgs2 r-elevation-1544tr9 r-flexDirection-18u37iz r-justifyContent-1777fci r-marginHorizontal-4amgru r-marginLeft-1wezhl r-marginTop-19inuqg r-marginVertical-y3xmzu r-width-sgscqh"
          >
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-eqz5dr r-justifyContent-1777fci r-marginHorizontal-1n9sb9w r-marginVertical-1uw6utc"
            >
              <div
                className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
                style={
                  Object {
                    "height": "180px",
                    "width": "200px",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                  style={
                    Object {
                      "backgroundImage": "url(\\"https://imgur.com/0r8qpuJ.png\\")",
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="https://imgur.com/0r8qpuJ.png"
                />
              </div>
              <div
                className="css-text-901oao r-color-jwli3a r-fontSize-h1js2l r-fontWeight-vw2c0b r-marginTop-1x0uki6"
                dir="auto"
                onClick={[Function]}
              >
                Coloring
              </div>
            </div>
          </div>
        </div>
        <div
          className="css-view-1dbjc4n r-cursor-1loqt21 r-touchAction-1otgn73 r-transitionProperty-1i6wzkk r-userSelect-lrvibr"
          data-focusable={true}
          onClick={[Function]}
          onContextMenu={[Function]}
          onKeyDown={[Function]}
          style={
            Object {
              "WebkitTransitionDuration": "0s",
              "transitionDuration": "0s",
            }
          }
          tabIndex="0"
        >
          <div
            className="css-view-1dbjc4n r-backgroundColor-19eppd8 r-borderRadius-kdyh1x r-boxShadow-l8kgs2 r-elevation-1544tr9 r-flexDirection-18u37iz r-justifyContent-1777fci r-marginHorizontal-4amgru r-marginLeft-1wezhl r-marginVertical-y3xmzu r-width-sgscqh"
          >
            <div
              className="css-view-1dbjc4n r-alignItems-1awozwy r-flexDirection-eqz5dr r-justifyContent-1777fci r-marginHorizontal-1n9sb9w r-marginVertical-1uw6utc"
            >
              <div
                className="css-view-1dbjc4n r-flexBasis-1mlwlqe r-overflow-1udh08x r-zIndex-417010"
                style={
                  Object {
                    "height": "180px",
                    "width": "200px",
                  }
                }
              >
                <div
                  className="css-view-1dbjc4n r-backgroundColor-1niwhzg r-backgroundPosition-vvn4in r-backgroundRepeat-u6sd8q r-backgroundSize-4gszlv r-bottom-1p0dtai r-height-1pi2tsx r-left-1d2f490 r-position-u8s1d r-right-zchlnj r-top-ipm5af r-width-13qz1uu r-zIndex-1wyyakw"
                  style={
                    Object {
                      "backgroundImage": "url(\\"https://imgur.com/Hsm0YR5.png\\")",
                    }
                  }
                />
                <img
                  alt=""
                  className="css-accessibilityImage-9pa8cd"
                  draggable={false}
                  src="https://imgur.com/Hsm0YR5.png"
                />
              </div>
              <div
                className="css-text-901oao r-color-jwli3a r-fontSize-h1js2l r-fontWeight-vw2c0b r-marginTop-1x0uki6"
                dir="auto"
                onClick={[Function]}
              >
                Drawing
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
